### PR TITLE
refactoring done on App, minor changes on Login and StoreContextProvider

### PR DESCRIPTION
**Pull request by DeepaAnil**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,32 +13,34 @@ function App() {
...
+    components = <><Overlay /><ErrorModel /></>;
+    elementId = 'errorModal';
+  } 
+  else if (storeCtx.isLoggedIn && !storeCtx.error && storeCtx.preLoader) {
+    components = <><Overlay /><Preloader /></>;
+    elementId = 'errorModal';
+  } 
+  else if (storeCtx.isLoggedIn && weatherctx.position !== "") {
+    components = <WheatherPage />;
+    elementId = "";
```
Comment (vk-detrans) [inline src/App.js:33]: Ignoring how it is done, do we need to defining it here?

File: src/App.js [hunk 2]
```diff
@@ -13,32 +13,34 @@ function App() {
...
-            <Preloader />
-          </>,
-          document.getElementById("errorModal")
-        )}
-      {!storeCtx.isLoggedIn &&
-        ReactDOM.createPortal(<Login />, document.getElementById("loginModal"))}
-      {storeCtx.isLoggedIn && weatherctx.position !== "" && <WheatherPage />}
+      {(elementId === "")
+        ? components
+        : ReactDOM.createPortal(components, document.getElementById(elementId))}
```
Comment (vk-detrans) [inline src/App.js:40]: Did you check why we need to create portal here?

File: src/Components/context/Store/StoreContextProvider.js [hunk 3]
```diff
@@ -17,9 +17,8 @@ export default function StoreContextProvider(props) {
 
   // Login Handler
   const logInHander = (name) => {
-    localStorage.setItem("userName", name);
```
Comment (vk-detrans) [inline src/Components/context/Store/StoreContextProvider.js:20]: Why was this removed from local storage? Doesn't it affect other parts of the app?

**Review by lingadurai98 — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (DeepaAnil) [reply src/App.js:33]: Initially, that conditional logic was inside the return statement. I just thought of moving it outside the return statement so that jsx code will be short and clean.

File: src/Components/context/Store/StoreContextProvider.js (hunk 3 above)
Comment (DeepaAnil) [reply src/Components/context/Store/StoreContextProvider.js:20]: I did not remove the local storage. It is found that the same lines of code exist in two functions. Hence reused that function instead.

File: src/App.js (hunk 2 above)
Comment (DeepaAnil) [reply src/App.js:40]: As per understanding, portal is used to render something to a child dom. Here root is used only for rendering weather information. Login, error etc are rendered to child doms', using createPortal.
